diff -u OSCAR-2.2.2/OSCAR-fct.py OSCAR/OSCAR-fct.py
--- OSCAR-2.2.2/OSCAR-fct.py	2018-07-11 10:13:19.000000000 +0200
+++ OSCAR/OSCAR-fct.py	2020-05-19 17:41:02.676086440 +0200
@@ -20,6 +20,8 @@
 
 
 import csv
+import time
+import copy
 
 import numpy as np
 import matplotlib.pyplot as plt
@@ -28,103 +30,37 @@
 
 
 ##################################################
-#   1. OSCAR LITE
+#   0. OSCAR LITE - one step
 ##################################################
 
-def OSCAR_lite(p=p,fT=fT,\
-               EFF=EFF,ECH4=ECH4,EN2O=EN2O,\
-               LUC=LUC,HARV=HARV,SHIFT=SHIFT,\
-               EHFC=EHFC,EPFC=EPFC,EODS=EODS,\
-               ENOX=ENOX,ECO=ECO,EVOC=EVOC,ESO2=ESO2,ENH3=ENH3,EOC=EOC,EBC=EBC,\
-               RFcon=RFcon,RFvolc=RFvolc,RFsolar=RFsolar,
-               force_CO2=False,force_GHG=False,force_halo=False,force_RF=False,force_RFs=False,force_clim=False,\
-               var_output=['ELUC','OSNK','LSNK','D_CO2','RF','D_gst'],\
-               plot=[]):
-
-    #===============
-    # A. DEFINITIONS
-    #===============
-
-    # plot variables
-    var_plot = []
-    if plot is 'all' or plot is 'CO2' or 'CO2' in plot:
-        var_plot += ['D_CO2','OSNK','LSNK','ELUC','D_AREA','D_npp','D_efire','D_fmort','D_rh1','D_fmet','D_rh2','D_FIN','D_FOUT','D_FCIRC','EFIRE_luc','FMORT_luc','RH1_luc','FMET_luc','RH2_luc','EHWP1_luc','EHWP2_luc','EHWP3_luc']
-    if plot is 'all' or plot is 'CH4' or 'CH4' in plot:
-        var_plot += ['D_CH4','D_OHSNK_CH4','D_HVSNK_CH4','D_XSNK_CH4','D_EWET','D_EBB_CH4']
-    if plot is 'all' or plot is 'N2O' or 'N2O' in plot:
-        var_plot += ['D_N2O','D_HVSNK_N2O','D_EBB_N2O']
-    if plot is 'all' or plot is 'O3' or 'O3' in plot:
-        var_plot += ['D_O3t','D_O3s','D_EESC','D_N2O_lag','D_gst']
-    if plot is 'all' or plot is 'AER' or 'AER' in plot:
-        var_plot += ['D_SO4','D_POA','D_BC','D_NO3','D_SOA','D_AERh','RF_SO4','RF_POA','RF_BC','RF_NO3','RF_SOA','RF_cloud']
-    if plot is 'all' or plot is 'clim' or 'clim' in plot:
-        var_plot += ['RF','D_gst','D_gyp','RF_CO2','RF_CH4','RF_H2Os','RF_N2O','RF_halo','RF_O3t','RF_O3s','RF_SO4','RF_POA','RF_BC','RF_NO3','RF_SOA','RF_cloud','RF_BCsnow','RF_LCC']
-
-    # save variables
-    var_timeseries = list(set(var_output)|set(var_plot))
-    for var in var_timeseries:
-        # global variables
-        if var in ['D_mld','D_dic','D_pH']\
-        or var in ['OSNK','LSNK','D_OHSNK_CH4','D_HVSNK_CH4','D_XSNK_CH4','D_HVSNK_N2O']\
-        or var in ['D_kOH','D_hv']\
-        or var in ['D_O3t','D_EESC','D_O3s','D_SO4','D_POA','D_BC','D_NO3','D_SOA','D_AERh']\
-        or var in ['D_CO2','D_CH4','D_CH4_lag','D_N2O','D_N2O_lag']\
-        or var in ['RF','RF_warm','RF_atm','RF_CO2','RF_CH4','RF_H2Os','RF_N2O','RF_halo','RF_O3t','RF_O3s','RF_SO4','RF_POA','RF_BC','RF_NO3','RF_SOA','RF_cloud','RF_BCsnow','RF_LCC']\
-        or var in ['D_gst','D_sst','D_gyp','D_OHC']:
-            exec(var+'_t = np.zeros([ind_final+1],dtype=dty)')
-        # (region) variables
-        if var in ['ELUC','D_AWET','D_EWET','D_ewet','D_EBB_CO2','D_EBB_CH4','D_EBB_N2O','D_EBB_NOX','D_EBB_CO','D_EBB_VOC','D_EBB_SO2','D_EBB_NH3','D_EBB_OC','D_EBB_BC','D_lst','D_lyp']:
-            exec(var+'_t = np.zeros([ind_final+1,nb_regionI],dtype=dty)')        
-        # (region)*(biome) variables
-        if var in ['D_AREA','D_npp','D_efire','D_fmort','D_rh1','D_fmet','D_rh2','D_cveg','D_csoil1','D_csoil2']:
-            exec(var+'_t = np.zeros([ind_final+1,nb_regionI,nb_biome],dtype=dty)')
-        # (region)*(biome)*(biome)*(age) variables
-        if var in ['EFIRE_luc','FMORT_luc','RH1_luc','FMET_luc','RH2_luc','EHWP1_luc','EHWP2_luc','EHWP3_luc']\
-        or var in ['CVEG_luc','CSOIL1_luc','CSOIL2_luc','CHWP1_luc','CHWP2_luc','CHWP3_luc']:
-            exec(var+'_t = np.zeros([ind_final+1,nb_regionI,nb_biome,nb_biome,ind_final+1],dtype=dty)')        
-        # (obox) variables
-        if var in ['D_FIN','D_FOUT','D_FCIRC','D_CSURF']:
-            exec(var+'_t = np.zeros([ind_final+1,nb_obox],dtype=dty)')
-        # (species) variables
-        if var in ['D_HFC','D_HFC_lag','D_OHSNK_HFC','D_HVSNK_HFC','D_XSNK_HFC']:
-            exec(var+'_t = np.zeros([ind_final+1,nb_HFC],dtype=dty)')
-        if var in ['D_PFC','D_PFC_lag','D_OHSNK_PFC','D_HVSNK_PFC','D_XSNK_PFC']:
-            exec(var+'_t = np.zeros([ind_final+1,nb_PFC],dtype=dty)')
-        if var in ['D_ODS','D_ODS_lag','D_OHSNK_ODS','D_HVSNK_ODS','D_XSNK_ODS']:
-            exec(var+'_t = np.zeros([ind_final+1,nb_ODS],dtype=dty)')
-
-    # run variables
-    # ocean
-    D_dic = np.array([0],dtype=dty)
-    D_CSURF = np.zeros([nb_obox],dtype=dty)
-    # land
-    for var in ['D_AREA','D_cveg','D_csoil1','D_csoil2']:
-        exec(var+' = np.zeros([nb_regionI,nb_biome],dtype=dty)')
-    # land-use
-    for var in ['CVEG_luc','CSOIL1_luc','CSOIL2_luc','CHWP1_luc','CHWP2_luc','CHWP3_luc']:
-        exec(var+' = np.zeros([nb_regionI,nb_biome,nb_biome,ind_final+1],dtype=dty)')
-    # atmosphere
-    for var in ['D_CO2','D_CH4','D_CH4_lag','D_N2O','D_N2O_lag','D_EESC','D_O3s']:
-        exec(var+' = np.array([0],dtype=dty)')
-    for var in ['D_HFC','D_HFC_lag','D_PFC','D_PFC_lag','D_ODS','D_ODS_lag']:
-        exec(var+' = np.zeros([nb_'+var[2:2+3]+'],dtype=dty)')
-    # climate
-    for var in ['D_gst','D_gst0','D_sst','D_gyp','D_OHC']:
-        exec(var+' = np.array([0],dtype=dty)')
-    for var in ['D_lst','D_lyp']:
-        exec(var+' = np.zeros([nb_regionI],dtype=dty)')
+# as input every variable of Oscar which is required 
+# and every Drivers (Exx) that are modified by Imaclim)
+def OSCAR_lite_onestep( p, fT, t_current, t_horizon, \
+EFF, ECH4, ECH4_fugi, EN2O, LUC, HARV, SHIFT, EHFC, EPFC, EODS,\
+ENOX, ECO, EVOC, ESO2, ENH3, EOC, EBC, \
+RFcon, RFvolc, RFsolar, force_CO2, force_GHG, force_halo, force_RF, force_RFs, force_clim, \
+D_CSURF, D_dic, \
+D_AREA, D_cveg, D_csoil1, D_csoil2, \
+CVEG_luc, CSOIL2_luc, CSOIL1_luc, CHWP1_luc, CHWP2_luc, CHWP3_luc, \
+D_EESC, D_O3s, \
+D_CH4_fugi, D_CO2_from_CH4, \
+D_CH4_fugi_lag,D_CH4_lag,D_N2O_lag,D_HFC_lag,D_PFC_lag,D_ODS_lag, \
+D_CO2, D_CH4, D_N2O, D_HFC, D_PFC, D_ODS, \
+D_gst, D_gst_CO2, D_gst_CH4, D_gst_SO4, D_gst_BC, D_gst0, D_gst0_CO2, D_gst0_CH4, D_gst0_SO4, D_gst0_BC, D_sst, D_lst, D_gyp, D_lyp, D_OHC, irf=''):
 
     #=======
     # B. RUN
     #=======
-    
-    for t in range(1,ind_final+1):
+    for t_step in range(t_horizon):
+        #t = t_current + (t_step+1)
+        t = t_current + (t_step)
         for tt in range(p):
 
             #---------
             # 1. OCEAN
             #---------
 
+
             # structure
             D_mld = mld_0 * alpha_mld * (np.exp(gamma_mld*fT*D_sst)-1)
             # fluxes
@@ -133,7 +69,8 @@
             D_FCIRC = D_CSURF * (1/tau_circ)
             OSNK = np.sum(D_FOUT - D_FIN)
             # stocks
-            D_CSURF += (p**-1) * (D_FIN - D_FOUT - D_FCIRC)
+            #D_CSURF += (p**-1) * (D_FIN - D_FOUT - D_FCIRC)
+            D_CSURF = D_CSURF + (p**-1) * (D_FIN - D_FOUT - D_FCIRC)
             D_dic = alpha_dic * np.sum(D_CSURF) / (1+D_mld/mld_0)
 
             #--------
@@ -141,7 +78,8 @@
             #--------
 
             # land-cover
-            D_AREA += (p**-1) * (np.sum(LUC[t],1) - np.sum(LUC[t],2))
+            #D_AREA += (p**-1) * (np.sum(LUC[t],1) - np.sum(LUC[t],2))
+            D_AREA = D_AREA + (p**-1) * (np.sum(LUC[t],1) - np.sum(LUC[t],2))
             D_AWET = AWET_0*(gamma_wetT*fT*D_lst + gamma_wetP*fT*D_lyp + gamma_wetC*fT*D_CO2)
             # factors
             D_k_igni = gamma_igniT*fT*D_lst[:,np.newaxis]+gamma_igniP*fT*D_lyp[:,np.newaxis]+gamma_igniC*fT*D_CO2
@@ -157,9 +95,12 @@
             LSNK = np.sum((D_rh1 + D_rh2 + D_efire - D_npp)*(AREA_0 + D_AREA))
             D_EWET = ewet_0*D_AWET + D_ewet*AWET_0 + D_ewet*D_AWET
             # stocks
-            D_cveg += (p**-1) * (D_npp - D_fmort - D_efire)
-            D_csoil1 += (p**-1) * (D_fmort - D_fmet - D_rh1)
-            D_csoil2 += (p**-1) * (D_fmet - D_rh2)
+            #D_cveg += (p**-1) * (D_npp - D_fmort - D_efire)
+            #D_csoil1 += (p**-1) * (D_fmort - D_fmet - D_rh1)
+            #D_csoil2 += (p**-1) * (D_fmet - D_rh2)
+            D_cveg = D_cveg + (p**-1) * (D_npp - D_fmort - D_efire)
+            D_csoil1 = D_csoil1 + (p**-1) * (D_fmort - D_fmet - D_rh1)
+            D_csoil2 = D_csoil2 + (p**-1) * (D_fmet - D_rh2)
 
             #------------
             # 3. LAND-USE
@@ -169,34 +110,55 @@
             # land-use change
             for b1 in range(nb_biome):
                 for b2 in range(nb_biome):
-                    CVEG_luc[:,b1,b2,t] +=  (p**-1) * -(cveg_0+D_cveg)[:,b2] * LUC[t,:,b1,b2] 
-                    CSOIL1_luc[:,b1,b2,t] += (p**-1) * ((csoil1_0+D_csoil1)[:,b1] - (csoil1_0+D_csoil1)[:,b2]) * LUC[t,:,b1,b2]
-                    CSOIL2_luc[:,b1,b2,t] += (p**-1) * ((csoil2_0+D_csoil2)[:,b1] - (csoil2_0+D_csoil2)[:,b2]) * LUC[t,:,b1,b2]
-                    CSOIL1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (p_AGB[:,b1]*p_HWP0[:,b1]+(1-p_AGB[:,b1])) * LUC[t,:,b1,b2]
-                    CHWP1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP1[:,b1] * LUC[t,:,b1,b2]
-                    CHWP2_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP2[:,b1] * LUC[t,:,b1,b2]
-                    CHWP3_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP3[:,b1] * LUC[t,:,b1,b2]
+                    #CVEG_luc[:,b1,b2,t] +=  (p**-1) * -(cveg_0+D_cveg)[:,b2] * LUC[t,:,b1,b2] 
+                    #CSOIL1_luc[:,b1,b2,t] += (p**-1) * ((csoil1_0+D_csoil1)[:,b1] - (csoil1_0+D_csoil1)[:,b2]) * LUC[t,:,b1,b2]
+                    #CSOIL2_luc[:,b1,b2,t] += (p**-1) * ((csoil2_0+D_csoil2)[:,b1] - (csoil2_0+D_csoil2)[:,b2]) * LUC[t,:,b1,b2]
+                    #CSOIL1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (p_AGB[:,b1]*p_HWP0[:,b1]+(1-p_AGB[:,b1])) * LUC[t,:,b1,b2]
+                    #CHWP1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP1[:,b1] * LUC[t,:,b1,b2]
+                    #CHWP2_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP2[:,b1] * LUC[t,:,b1,b2]
+                    #CHWP3_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP3[:,b1] * LUC[t,:,b1,b2]
+                    CVEG_luc[:,b1,b2,t] = CVEG_luc[:,b1,b2,t] +  (p**-1) * -(cveg_0+D_cveg)[:,b2] * LUC[t,:,b1,b2] 
+                    CSOIL1_luc[:,b1,b2,t] = CSOIL1_luc[:,b1,b2,t] + (p**-1) * ((csoil1_0+D_csoil1)[:,b1] - (csoil1_0+D_csoil1)[:,b2]) * LUC[t,:,b1,b2]
+                    CSOIL2_luc[:,b1,b2,t] = CSOIL2_luc[:,b1,b2,t] + (p**-1) * ((csoil2_0+D_csoil2)[:,b1] - (csoil2_0+D_csoil2)[:,b2]) * LUC[t,:,b1,b2]
+                    CSOIL1_luc[:,b1,b2,t] = CSOIL1_luc[:,b1,b2,t] + (p**-1) * (cveg_0+D_cveg)[:,b1] * (p_AGB[:,b1]*p_HWP0[:,b1]+(1-p_AGB[:,b1])) * LUC[t,:,b1,b2]
+                    CHWP1_luc[:,b1,b2,t] = CHWP1_luc[:,b1,b2,t]  + (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP1[:,b1] * LUC[t,:,b1,b2]
+                    CHWP2_luc[:,b1,b2,t] = CHWP2_luc[:,b1,b2,t] + (p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP2[:,b1] * LUC[t,:,b1,b2]
+                    CHWP3_luc[:,b1,b2,t] = CHWP3_luc[:,b1,b2,t] +(p**-1) * (cveg_0+D_cveg)[:,b1] * p_AGB[:,b1]*p_HWP3[:,b1] * LUC[t,:,b1,b2]
             # harvest
             for b in range(nb_biome):
-                CVEG_luc[:,b,b,t] += (p**-1) * -HARV[t,:,b]
-                CSOIL1_luc[:,b,b,t] += (p**-1) * p_HWP0[:,b] * HARV[t,:,b]
-                CHWP1_luc[:,b,b,t] += (p**-1) * p_HWP1[:,b] * HARV[t,:,b]
-                CHWP2_luc[:,b,b,t] += (p**-1) * p_HWP2[:,b] * HARV[t,:,b]
-                CHWP3_luc[:,b,b,t] += (p**-1) * p_HWP3[:,b] * HARV[t,:,b]
+                #CVEG_luc[:,b,b,t] += (p**-1) * -HARV[t,:,b]
+                #CSOIL1_luc[:,b,b,t] += (p**-1) * p_HWP0[:,b] * HARV[t,:,b]
+                #CHWP1_luc[:,b,b,t] += (p**-1) * p_HWP1[:,b] * HARV[t,:,b]
+                #CHWP2_luc[:,b,b,t] += (p**-1) * p_HWP2[:,b] * HARV[t,:,b]
+                #CHWP3_luc[:,b,b,t] += (p**-1) * p_HWP3[:,b] * HARV[t,:,b]
+                CVEG_luc[:,b,b,t] = CVEG_luc[:,b,b,t] + (p**-1) * -HARV[t,:,b]
+                CSOIL1_luc[:,b,b,t] = CSOIL1_luc[:,b,b,t] + (p**-1) * p_HWP0[:,b] * HARV[t,:,b]
+                CHWP1_luc[:,b,b,t] = CHWP1_luc[:,b,b,t] + (p**-1) * p_HWP1[:,b] * HARV[t,:,b]
+                CHWP2_luc[:,b,b,t] = CHWP2_luc[:,b,b,t] + (p**-1) * p_HWP2[:,b] * HARV[t,:,b]
+                CHWP3_luc[:,b,b,t] = CHWP3_luc[:,b,b,t] + (p**-1) * p_HWP3[:,b] * HARV[t,:,b]
             # shifting cultivation
             for b1 in range(nb_biome):
                 for b2 in range(b1,nb_biome):
-                    CVEG_luc[:,b1,b2,t] += (p**-1) * -(cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * SHIFT[t,:,b1,b2]
-                    CSOIL1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * (p_AGB[:,b1]*p_HWP0[:,b1]+(1-p_AGB[:,b1])) * SHIFT[t,:,b1,b2]
-                    CHWP1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP1[:,b1] * SHIFT[t,:,b1,b2]
-                    CHWP2_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP2[:,b1] * SHIFT[t,:,b1,b2]
-                    CHWP3_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP3[:,b1] * SHIFT[t,:,b1,b2]
-                    CVEG_luc[:,b2,b1,t] += (p**-1) * -(cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * SHIFT[t,:,b1,b2]
-                    CSOIL1_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * (p_AGB[:,b2]*p_HWP0[:,b2]+(1-p_AGB[:,b2])) * SHIFT[t,:,b1,b2]
-                    CHWP1_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP1[:,b2] * SHIFT[t,:,b1,b2]
-                    CHWP2_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP2[:,b2] * SHIFT[t,:,b1,b2]
-                    CHWP3_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP3[:,b2] * SHIFT[t,:,b1,b2]
-
+                    #CVEG_luc[:,b1,b2,t] += (p**-1) * -(cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * SHIFT[t,:,b1,b2]
+                    #CSOIL1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * (p_AGB[:,b1]*p_HWP0[:,b1]+(1-p_AGB[:,b1])) * SHIFT[t,:,b1,b2]
+                    #CHWP1_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP1[:,b1] * SHIFT[t,:,b1,b2]
+                    #CHWP2_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP2[:,b1] * SHIFT[t,:,b1,b2]
+                    #CHWP3_luc[:,b1,b2,t] += (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP3[:,b1] * SHIFT[t,:,b1,b2]
+                    #CVEG_luc[:,b2,b1,t] += (p**-1) * -(cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * SHIFT[t,:,b1,b2]
+                    #CSOIL1_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * (p_AGB[:,b2]*p_HWP0[:,b2]+(1-p_AGB[:,b2])) * SHIFT[t,:,b1,b2]
+                    #CHWP1_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP1[:,b2] * SHIFT[t,:,b1,b2]
+                    #CHWP2_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP2[:,b2] * SHIFT[t,:,b1,b2]
+                    #CHWP3_luc[:,b2,b1,t] += (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP3[:,b2] * SHIFT[t,:,b1,b2]
+                    CVEG_luc[:,b1,b2,t] = CVEG_luc[:,b1,b2,t] + (p**-1) * -(cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * SHIFT[t,:,b1,b2]
+                    CSOIL1_luc[:,b1,b2,t] = CSOIL1_luc[:,b1,b2,t] + (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * (p_AGB[:,b1]*p_HWP0[:,b1]+(1-p_AGB[:,b1])) * SHIFT[t,:,b1,b2]
+                    CHWP1_luc[:,b1,b2,t] = CHWP1_luc[:,b1,b2,t] + (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP1[:,b1] * SHIFT[t,:,b1,b2]
+                    CHWP2_luc[:,b1,b2,t] = CHWP2_luc[:,b1,b2,t] + (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP2[:,b1] * SHIFT[t,:,b1,b2]
+                    CHWP3_luc[:,b1,b2,t] = CHWP3_luc[:,b1,b2,t] + (p**-1) * (cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * p_AGB[:,b1]*p_HWP3[:,b1] * SHIFT[t,:,b1,b2]
+                    CVEG_luc[:,b2,b1,t] = CVEG_luc[:,b2,b1,t] + (p**-1) * -(cveg_0+D_cveg)[:,b1] * (1-np.exp(-mu_0[:,b1]*tau_shift)) * SHIFT[t,:,b1,b2]
+                    CSOIL1_luc[:,b2,b1,t] = CSOIL1_luc[:,b2,b1,t] + (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * (p_AGB[:,b2]*p_HWP0[:,b2]+(1-p_AGB[:,b2])) * SHIFT[t,:,b1,b2]
+                    CHWP1_luc[:,b2,b1,t] = CHWP1_luc[:,b2,b1,t] + (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP1[:,b2] * SHIFT[t,:,b1,b2]
+                    CHWP2_luc[:,b2,b1,t] = CHWP2_luc[:,b2,b1,t] + (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP2[:,b2] * SHIFT[t,:,b1,b2]
+                    CHWP3_luc[:,b2,b1,t] = CHWP3_luc[:,b2,b1,t] + (p**-1) * (cveg_0+D_cveg)[:,b2] * (1-np.exp(-mu_0[:,b2]*tau_shift)) * p_AGB[:,b2]*p_HWP3[:,b2] * SHIFT[t,:,b1,b2]
             # fluxes
             # book-keeping model
             NPP_luc = 0*CVEG_luc
@@ -215,16 +177,24 @@
             # biomass burning
             for VAR in ['CO2','CH4','N2O','NOX','CO','VOC','SO2','NH3','OC','BC']:
                 exec('D_EBB_'+VAR+' = np.sum( alpha_BB_'+VAR+'*(igni_0*cveg_0*D_AREA + D_efire*AREA_0 + D_efire*D_AREA) ,1)')
-                exec('D_EBB_'+VAR+' += np.sum(np.sum(np.sum( alpha_BB_'+VAR+'[:,:,np.newaxis,np.newaxis]*EHWP1_luc ,3),2),1)')
-                exec('D_EBB_'+VAR+' += np.sum(np.sum(np.sum( alpha_BB_'+VAR+'[:,np.newaxis,:,np.newaxis]*EFIRE_luc ,3),2),1)')
+                #exec('D_EBB_'+VAR+' += np.sum(np.sum(np.sum( alpha_BB_'+VAR+'[:,:,np.newaxis,np.newaxis]*EHWP1_luc ,3),2),1)')
+                #exec('D_EBB_'+VAR+' += np.sum(np.sum(np.sum( alpha_BB_'+VAR+'[:,np.newaxis,:,np.newaxis]*EFIRE_luc ,3),2),1)')
+                exec('D_EBB_'+VAR+' = D_EBB_'+VAR+' + np.sum(np.sum(np.sum( alpha_BB_'+VAR+'[:,:,np.newaxis,np.newaxis]*EHWP1_luc ,3),2),1)')
+                exec('D_EBB_'+VAR+' = D_EBB_'+VAR+' + np.sum(np.sum(np.sum( alpha_BB_'+VAR+'[:,np.newaxis,:,np.newaxis]*EFIRE_luc ,3),2),1)')
            
             # stocks
-            CVEG_luc += (p**-1) * (NPP_luc - FMORT_luc - EFIRE_luc)
-            CSOIL1_luc += (p**-1) * (FMORT_luc - FMET_luc - RH1_luc)
-            CSOIL2_luc += (p**-1) * (FMET_luc - RH2_luc)
-            CHWP1_luc += (p**-1) * -EHWP1_luc
-            CHWP2_luc += (p**-1) * -EHWP2_luc
-            CHWP3_luc += (p**-1) * -EHWP3_luc
+            #CVEG_luc += (p**-1) * (NPP_luc - FMORT_luc - EFIRE_luc)
+            #CSOIL1_luc += (p**-1) * (FMORT_luc - FMET_luc - RH1_luc)
+            #CSOIL2_luc += (p**-1) * (FMET_luc - RH2_luc)
+            #CHWP1_luc += (p**-1) * -EHWP1_luc
+            #CHWP2_luc += (p**-1) * -EHWP2_luc
+            #CHWP3_luc += (p**-1) * -EHWP3_luc
+            CVEG_luc = CVEG_luc + (p**-1) * (NPP_luc - FMORT_luc - EFIRE_luc)
+            CSOIL1_luc = CSOIL1_luc + (p**-1) * (FMORT_luc - FMET_luc - RH1_luc)
+            CSOIL2_luc = CSOIL2_luc + (p**-1) * (FMET_luc - RH2_luc)
+            CHWP1_luc = CHWP1_luc + (p**-1) * -EHWP1_luc
+            CHWP2_luc = CHWP2_luc + (p**-1) * -EHWP2_luc
+            CHWP3_luc = CHWP3_luc + (p**-1) * -EHWP3_luc
             
             #-------------
             # 4. CHEMISTRY
@@ -237,6 +207,10 @@
             D_OHSNK_CH4 = -alpha_CH4/tau_CH4_OH * (CH4_0*D_kOH + D_CH4 + D_kOH*D_CH4)
             D_HVSNK_CH4 = -alpha_CH4/tau_CH4_hv * (CH4_0*D_hv + D_CH4_lag + D_hv*D_CH4_lag)
             D_XSNK_CH4 = -alpha_CH4*(1/tau_CH4_soil + 1/tau_CH4_ocean) * D_CH4
+            D_OHSNK_CH4_fugi = -alpha_CH4/tau_CH4_OH * (CH4_0*D_kOH + D_CH4_fugi + D_kOH*D_CH4_fugi)
+            D_HVSNK_CH4_fugi = -alpha_CH4/tau_CH4_hv * (CH4_0*D_hv + D_CH4_fugi_lag + D_hv*D_CH4_fugi_lag)
+            D_XSNK_CH4_fugi = -alpha_CH4*(1/tau_CH4_soil + 1/tau_CH4_ocean) * D_CH4_fugi
+            D_SNK_CH4_fugi = D_OHSNK_CH4_fugi + D_HVSNK_CH4_fugi + D_XSNK_CH4_fugi
             D_HVSNK_N2O = -alpha_N2O/tau_N2O_hv * (N2O_0*D_hv + D_N2O_lag + D_hv*D_N2O_lag)
             for VAR in ['HFC','PFC','ODS']:
                 exec('D_OHSNK_'+VAR+' = -alpha_'+VAR+'/tau_'+VAR+'_OH * ('+VAR+'_0*D_kOH + D_'+VAR+' + D_kOH*D_'+VAR+')')
@@ -244,9 +218,12 @@
                 exec('D_XSNK_'+VAR+' = -alpha_'+VAR+'/tau_'+VAR+'_othr * D_'+VAR)
             # stocks
             D_O3t = chi_O3t_CH4*np.log(1+D_CH4/CH4_0) + Gamma_O3t*fT*D_gst
-            D_O3t += chi_O3t_NOX*np.sum(w_reg_NOX*np.sum(p_reg4*(ENOX[t]+D_EBB_NOX)[:,np.newaxis],0))
-            D_O3t += chi_O3t_CO*np.sum(w_reg_CO*np.sum(p_reg4*(ECO[t]+D_EBB_CO)[:,np.newaxis],0))
-            D_O3t += chi_O3t_VOC*np.sum(w_reg_VOC*np.sum(p_reg4*(EVOC[t]+D_EBB_VOC)[:,np.newaxis],0))
+            #D_O3t += chi_O3t_NOX*np.sum(w_reg_NOX*np.sum(p_reg4*(ENOX[t]+D_EBB_NOX)[:,np.newaxis],0))
+            #D_O3t += chi_O3t_CO*np.sum(w_reg_CO*np.sum(p_reg4*(ECO[t]+D_EBB_CO)[:,np.newaxis],0))
+            #D_O3t += chi_O3t_VOC*np.sum(w_reg_VOC*np.sum(p_reg4*(EVOC[t]+D_EBB_VOC)[:,np.newaxis],0))
+            D_O3t = D_O3t + chi_O3t_NOX*np.sum(w_reg_NOX*np.sum(p_reg4*(ENOX[t]+D_EBB_NOX)[:,np.newaxis],0))
+            D_O3t = D_O3t + chi_O3t_CO*np.sum(w_reg_CO*np.sum(p_reg4*(ECO[t]+D_EBB_CO)[:,np.newaxis],0))
+            D_O3t = D_O3t + chi_O3t_VOC*np.sum(w_reg_VOC*np.sum(p_reg4*(EVOC[t]+D_EBB_VOC)[:,np.newaxis],0))
             D_EESC = np.sum(f_fracrel(tau_lag) * (n_Cl+alpha_Br*n_Br) * D_ODS_lag)
             D_O3s = chi_O3s_EESC*D_EESC + chi_O3s_N2O*D_N2O_lag * (1-D_EESC/EESC_x) + Gamma_O3s*fT*D_gst
             D_SO4 = alpha_SO4*tau_SO2*np.sum(w_reg_SO2*np.sum(p_reg4*(ESO2[t]+D_EBB_SO2)[:,np.newaxis],0)) + alpha_SO4*tau_DMS*0 + Gamma_SO4*fT*D_gst
@@ -263,14 +240,25 @@
             #--------------
 
             # stocks
-            D_CO2 += (p**-1) * (1/alpha_CO2) * (np.sum(EFF[t]) + np.sum(ELUC) + LSNK + OSNK)
-            D_CH4 += (p**-1) * (1/alpha_CH4) * (np.sum(ECH4[t]) + np.sum(D_EBB_CH4) + np.sum(D_EWET) + D_OHSNK_CH4 + D_HVSNK_CH4 + D_XSNK_CH4)
-            D_N2O += (p**-1) * (1/alpha_N2O) * (np.sum(EN2O[t]) + np.sum(D_EBB_N2O) + D_HVSNK_N2O)
-            D_HFC += (p**-1) * (1/alpha_HFC) * (np.sum(EHFC[t],0) + D_OHSNK_HFC + D_HVSNK_HFC + D_XSNK_HFC)
-            D_PFC += (p**-1) * (1/alpha_PFC) * (np.sum(EPFC[t],0) + D_OHSNK_PFC + D_HVSNK_PFC + D_XSNK_PFC)
-            D_ODS += (p**-1) * (1/alpha_ODS) * (np.sum(EODS[t],0) + D_OHSNK_ODS + D_HVSNK_ODS + D_XSNK_ODS)
-            for VAR in ['CH4','N2O','HFC','PFC','ODS']:
-                exec('D_'+VAR+'_lag += (p**-1) * ((1/tau_lag)*D_'+VAR+' - (1/tau_lag)*D_'+VAR+'_lag)')
+            #D_CH4_fugi += (p**-1) * (1/alpha_CH4) * (np.sum(ECH4_fugi[t]) + D_SNK_CH4_fugi)
+            #D_CO2_from_CH4 += (p**-1) * (1/alpha_CO2) * ( - D_SNK_CH4_fugi * 1e-3)
+            D_CH4_fugi = D_CH4_fugi + (p**-1) * (1/alpha_CH4) * (np.sum(ECH4_fugi[t]) + D_SNK_CH4_fugi)
+            D_CO2_from_CH4 = D_CO2_from_CH4 + (p**-1) * (1/alpha_CO2) * ( - D_SNK_CH4_fugi * 1e-3)
+            #D_CO2 += (p**-1) * (1/alpha_CO2) * (np.sum(EFF[t]) + np.sum(ELUC) + LSNK + OSNK - D_SNK_CH4_fugi * 1e-3)
+            D_CO2 = D_CO2 + (p**-1) * (1/alpha_CO2) * (np.sum(EFF[t]) + np.sum(ELUC) + LSNK + OSNK - D_SNK_CH4_fugi * 1e-3)
+            #D_CH4 += (p**-1) * (1/alpha_CH4) * (np.sum(ECH4[t]) + np.sum(D_EBB_CH4) + np.sum(D_EWET) + D_OHSNK_CH4 + D_HVSNK_CH4 + D_XSNK_CH4)
+            #D_N2O += (p**-1) * (1/alpha_N2O) * (np.sum(EN2O[t]) + np.sum(D_EBB_N2O) + D_HVSNK_N2O)
+            #D_HFC += (p**-1) * (1/alpha_HFC) * (np.sum(EHFC[t],0) + D_OHSNK_HFC + D_HVSNK_HFC + D_XSNK_HFC)
+            #D_PFC += (p**-1) * (1/alpha_PFC) * (np.sum(EPFC[t],0) + D_OHSNK_PFC + D_HVSNK_PFC + D_XSNK_PFC)
+            #D_ODS += (p**-1) * (1/alpha_ODS) * (np.sum(EODS[t],0) + D_OHSNK_ODS + D_HVSNK_ODS + D_XSNK_ODS)
+            D_CH4 = D_CH4 + (p**-1) * (1/alpha_CH4) * (np.sum(ECH4[t]) + np.sum(D_EBB_CH4) + np.sum(D_EWET) + D_OHSNK_CH4 + D_HVSNK_CH4 + D_XSNK_CH4)
+            D_N2O = D_N2O + (p**-1) * (1/alpha_N2O) * (np.sum(EN2O[t]) + np.sum(D_EBB_N2O) + D_HVSNK_N2O)
+            D_HFC = D_HFC + (p**-1) * (1/alpha_HFC) * (np.sum(EHFC[t],0) + D_OHSNK_HFC + D_HVSNK_HFC + D_XSNK_HFC)
+            D_PFC = D_PFC + (p**-1) * (1/alpha_PFC) * (np.sum(EPFC[t],0) + D_OHSNK_PFC + D_HVSNK_PFC + D_XSNK_PFC)
+            D_ODS = D_ODS + (p**-1) * (1/alpha_ODS) * (np.sum(EODS[t],0) + D_OHSNK_ODS + D_HVSNK_ODS + D_XSNK_ODS)
+            for VAR in ['CH4_fugi','CH4','N2O','HFC','PFC','ODS']:
+                #exec('D_'+VAR+'_lag += (p**-1) * ((1/tau_lag)*D_'+VAR+' - (1/tau_lag)*D_'+VAR+'_lag)')
+                exec('D_'+VAR+'_lag = D_'+VAR+'_lag + (p**-1) * ((1/tau_lag)*D_'+VAR+' - (1/tau_lag)*D_'+VAR+'_lag)')
 
             # FORCE
             if force_CO2:
@@ -293,6 +281,7 @@
             # fluxes
             # per component
             RF_CO2 = f_RF_CO2(D_CO2)
+            RF_CO2_from_CH4 = f_RF_CO2(D_CO2_from_CH4) 
             RF_CH4 = f_RF_CH4(D_CH4)-(f_RF_overlap(D_CH4,D_N2O)-f_RF_overlap(0,D_N2O))
             RF_H2Os = f_RF_H2Os(D_CH4_lag)
             RF_N2O = f_RF_N2O(D_N2O)-(f_RF_overlap(D_CH4,D_N2O)-f_RF_overlap(D_CH4,0))
@@ -311,6 +300,7 @@
             # totals
             RF = RF_CO2 + RF_CH4 + RF_H2Os + RF_N2O + RF_halo + RF_O3t + RF_O3s + RF_SO4 + RF_POA + RF_BC + RF_NO3 + RF_SOA + RF_DUST + RF_SALT + RF_cloud + RF_BCsnow + RF_LCC + RFcon[t] + RFvolc[t] + RFsolar[t]
             RF_warm = RF_CO2 + RF_CH4 + RF_H2Os + RF_N2O + RF_halo + RF_O3t + RF_O3s + RF_SO4 + RF_POA + RF_BC + RF_NO3 + RF_SOA + RF_DUST + RF_SALT + RF_cloud + warmeff_BCsnow*RF_BCsnow + warmeff_LCC*RF_LCC + RFcon[t] + warmeff_volc*RFvolc[t] + RFsolar[t]
+            RF_warm_noLUC = RF_warm - warmeff_LCC*RF_LCC
             RF_atm = p_atm_CO2*RF_CO2 + p_atm_noCO2*(RF_CH4+RF_N2O+RF_halo) + p_atm_O3t*RF_O3t + p_atm_strat*(RF_O3s+RF_H2Os) + p_atm_scatter*(RF_SO4+RF_POA+RF_NO3+RF_SOA+RF_DUST+RF_SALT+RFvolc[t]) + p_atm_absorb*RF_BC + p_atm_cloud*(RF_cloud+RFcon[t]) + p_atm_alb*(RF_BCsnow+RF_LCC) + p_atm_solar*RFsolar[t]
 
             # FORCE
@@ -321,15 +311,34 @@
 
             # stocks
             # temperatures
-            D_gst += (p**-1) * (1/tau_gst) * (lambda_0*RF_warm - D_gst - theta_0*(D_gst-D_gst0))
-            D_gst0 += (p**-1) * (1/tau_gst0) * theta_0*(D_gst-D_gst0)
+            #D_gst += (p**-1) * (1/tau_gst) * (lambda_0*RF_warm - D_gst - theta_0*(D_gst-D_gst0))
+            #D_gst_CO2 += (p**-1) * (1/tau_gst) * (lambda_0*RF_CO2 - D_gst_CO2 - theta_0*(D_gst_CO2-D_gst0_CO2))
+            #D_gst_CH4 += (p**-1) * (1/tau_gst) * (lambda_0*RF_CH4 - D_gst_CH4 - theta_0*(D_gst_CH4-D_gst0_CH4))
+            #D_gst_SO4 += (p**-1) * (1/tau_gst) * (lambda_0*RF_SO4 - D_gst_SO4 - theta_0*(D_gst_SO4-D_gst0_SO4))
+            #D_gst_BC += (p**-1) * (1/tau_gst) * (lambda_0*(RF_BC+warmeff_BCsnow*RF_BCsnow) - D_gst_BC - theta_0*(D_gst_BC-D_gst0_BC))
+            #D_gst0 += (p**-1) * (1/tau_gst0) * theta_0*(D_gst-D_gst0)
+            #D_gst0_CO2 += (p**-1) * (1/tau_gst0) * theta_0*(D_gst_CO2-D_gst0_CO2)
+            #D_gst0_CH4 += (p**-1) * (1/tau_gst0) * theta_0*(D_gst_CH4-D_gst0_CH4)
+            #D_gst0_SO4 += (p**-1) * (1/tau_gst0) * theta_0*(D_gst_SO4-D_gst0_SO4)
+            #D_gst0_BC += (p**-1) * (1/tau_gst0) * theta_0*(D_gst_BC-D_gst0_BC)
+            D_gst = D_gst + (p**-1) * (1/tau_gst) * (lambda_0*RF_warm - D_gst - theta_0*(D_gst-D_gst0))
+            D_gst_CO2 = D_gst_CO2 + (p**-1) * (1/tau_gst) * (lambda_0*RF_CO2 - D_gst_CO2 - theta_0*(D_gst_CO2-D_gst0_CO2))
+            D_gst_CH4 = D_gst_CH4 + (p**-1) * (1/tau_gst) * (lambda_0*RF_CH4 - D_gst_CH4 - theta_0*(D_gst_CH4-D_gst0_CH4))
+            D_gst_SO4 = D_gst_SO4 + (p**-1) * (1/tau_gst) * (lambda_0*RF_SO4 - D_gst_SO4 - theta_0*(D_gst_SO4-D_gst0_SO4))
+            D_gst_BC = D_gst_BC + (p**-1) * (1/tau_gst) * (lambda_0*(RF_BC+warmeff_BCsnow*RF_BCsnow) - D_gst_BC - theta_0*(D_gst_BC-D_gst0_BC))
+            D_gst0 = D_gst0 + (p**-1) * (1/tau_gst0) * theta_0*(D_gst-D_gst0)
+            D_gst0_CO2 = D_gst0_CO2 + (p**-1) * (1/tau_gst0) * theta_0*(D_gst_CO2-D_gst0_CO2)
+            D_gst0_CH4 = D_gst0_CH4 + (p**-1) * (1/tau_gst0) * theta_0*(D_gst_CH4-D_gst0_CH4)
+            D_gst0_SO4 = D_gst0_SO4 + (p**-1) * (1/tau_gst0) * theta_0*(D_gst_SO4-D_gst0_SO4)
+            D_gst0_BC = D_gst0_BC + (p**-1) * (1/tau_gst0) * theta_0*(D_gst_BC-D_gst0_BC)
             D_sst = w_reg_sst*D_gst
             D_lst = w_reg_lst*D_gst
             # precipitations
             D_gyp = alpha_gyp*D_gst + beta_gyp*RF_atm
             D_lyp = w_reg_lyp*D_gyp
             # ocean
-            D_OHC += (p**-1) * p_OHC * alpha_OHC * (RF - D_gst/lambda_0)
+            #D_OHC += (p**-1) * p_OHC * alpha_OHC * (RF - D_gst/lambda_0)
+            D_OHC = D_OHC + (p**-1) * p_OHC * alpha_OHC * (RF - D_gst/lambda_0)
             D_pH = f_pH(D_CO2)
 
             # FORCE
@@ -342,14 +351,14 @@
             #-----------
             # Y. SAVE
             #-----------
-            
-            for var in var_timeseries:
-                exec(var+'_t[t] += (p**-1) * '+var)
+
+            #for var in var_timeseries:
+            #    exec(var+'_t[t] += (p**-1) * '+var)
 
             #---------
             # Z. TESTS
             #---------
-        
+
             if np.isnan(np.sum(D_CO2)):
                 print 'D_CO2 = NaN at t = '+str(t)+' and tt = '+str(tt)
                 print 'OSNK = '+str(np.sum(OSNK))
@@ -357,7 +366,7 @@
                 print 'ELUC = '+str(np.sum(ELUC))
                 break
             if np.isnan(np.sum(D_CH4)):
-                print 'D_CH4 = NaN at t = '+str(t)+' and tt = '+str(tt)              
+                print 'D_CH4 = NaN at t = '+str(t)+' and tt = '+str(tt)
                 print 'D_EWET = '+str(np.sum(D_EWET))
                 print 'D_OHSNK = '+str(np.sum(D_OHSNK_CH4))
                 print 'D_HVSNK = '+str(np.sum(D_HVSNK_CH4))
@@ -382,12 +391,275 @@
                 print 'RF_BCsnow = '+str(np.sum(RF_BCsnow))
                 print 'RF_LCC = '+str(np.sum(RF_LCC))
                 break
+             #print '    ', D_gst
+        D_C_LUC = (CVEG_luc + CSOIL1_luc + CSOIL2_luc + CHWP1_luc + CHWP2_luc + CHWP3_luc).sum()
+        if t_step==0:
+            RF_warm_list = list(RF_warm)
+            RF_warm_noLUC_list = list(RF_warm_noLUC)
+            D_CO2_list = list(D_CO2)
+            D_CLUC_list = list()
+            D_CLUC_list.append(D_C_LUC)
+        else:
+            RF_warm_list.append(RF_warm)
+            RF_warm_noLUC_list.append( RF_warm_noLUC)
+            D_CO2_list.append(D_CO2)
+            D_CLUC_list.append(D_C_LUC)
+    #print t, RF_warm, RF, D_gst
+    #print 'end: ', D_gst   
+    if irf != '':
+        np.savetxt(path_dialog_withImaclim+"/oscar"+irf+"_rfwarm", np.array(RF_warm_list), delimiter="|")
+        if 'gra_for' in irf or 'ref' in irf or 'CO2' in irf:
+            np.savetxt(path_dialog_withImaclim+"/oscar"+irf+"_rfwarmnoluc", np.array(RF_warm_noLUC_list), delimiter="|")
+            np.savetxt(path_dialog_withImaclim+"/oscar"+irf+"_dcO2", np.array(D_CO2_list), delimiter="|")
+            np.savetxt(path_dialog_withImaclim+"/oscar"+irf+"_dcluc", np.array(D_CLUC_list), delimiter="|")
+    else:    
+        return D_FIN, D_FOUT, D_FCIRC, OSNK, D_CSURF, D_dic, D_AREA, D_AWET, D_k_igni, D_k_rho, D_npp, D_efire, D_fmort, D_rh1, D_fmet, D_rh2, D_ewet, LSNK, D_EWET, D_cveg, D_csoil1, D_csoil2, \
+CVEG_luc, CSOIL2_luc, CSOIL1_luc, CHWP1_luc, CHWP2_luc, CHWP3_luc, \
+NPP_luc, EFIRE_luc, FMORT_luc, RH1_luc, FMET_luc, RH2_luc, EHWP1_luc, EHWP2_luc, EHWP3_luc, ELUC, \
+D_EBB_CO2,D_EBB_CH4,D_EBB_N2O,D_EBB_NOX,D_EBB_CO,D_EBB_VOC,D_EBB_SO2,D_EBB_NH3,D_EBB_OC,D_EBB_BC, \
+D_kOH, D_hv, D_OHSNK_CH4, D_HVSNK_CH4, D_XSNK_CH4, D_OHSNK_CH4_fugi, D_HVSNK_CH4_fugi, D_XSNK_CH4_fugi, D_SNK_CH4_fugi, D_HVSNK_N2O, \
+D_OHSNK_HFC, D_OHSNK_PFC, D_OHSNK_ODS, \
+D_HVSNK_HFC, D_HVSNK_PFC, D_HVSNK_ODS, \
+D_XSNK_HFC, D_XSNK_PFC, D_XSNK_ODS, \
+D_O3t, D_EESC, D_O3s, D_SO4, D_POA, D_BC, D_NO3, D_SOA, D_DUST, D_SALT, D_AERh, \
+D_CH4_fugi, D_CO2_from_CH4, \
+D_CH4_fugi_lag,D_CH4_lag,D_N2O_lag,D_HFC_lag,D_PFC_lag,D_ODS_lag, \
+D_CO2, D_CH4, D_N2O, D_HFC, D_PFC, D_ODS, \
+RF_CO2, RF_CO2_from_CH4, RF_CH4, RF_H2Os, RF_N2O, RF_halo, \
+RF_O3t,RF_O3s,RF_SO4,RF_POA,RF_BC,RF_NO3,RF_SOA,RF_DUST,RF_SALT, \
+RF_cloud, RF_BCsnow, RF_LCC, \
+RF, RF_warm, RF_atm, \
+D_gst, D_gst_CO2, D_gst_CH4, D_gst_SO4, D_gst_BC, D_gst0, D_gst0_CO2, D_gst0_CH4, D_gst0_SO4, D_gst0_BC, D_sst, D_lst, D_gyp, D_lyp, D_OHC, D_pH
 
+
+##################################################
+#   1. OSCAR LITE
+##################################################
+
+def OSCAR_lite(p=p,fT=fT,\
+               EFF=EFF,ECH4=ECH4,ECH4_fugi=np.zeros([ind_final+1,1],dtype=dty),EN2O=EN2O,\
+               LUC=LUC,HARV=HARV,SHIFT=SHIFT,\
+               EHFC=EHFC,EPFC=EPFC,EODS=EODS,\
+               ENOX=ENOX,ECO=ECO,EVOC=EVOC,ESO2=ESO2,ENH3=ENH3,EOC=EOC,EBC=EBC,\
+               RFcon=RFcon,RFvolc=RFvolc,RFsolar=RFsolar,
+               force_CO2=False,force_GHG=False,force_halo=False,force_RF=False,force_RFs=False,force_clim=False,\
+               var_output=['ELUC','OSNK','LSNK','D_CO2','RF','D_gst'],\
+               plot=[]):
+
+    #===============
+    # A. DEFINITIONS
+    #===============
+
+    # plot variables
+    var_plot = []
+    if plot is 'all' or plot is 'CO2' or 'CO2' in plot:
+        var_plot += ['D_CO2','OSNK','LSNK','ELUC','D_AREA','D_npp','D_efire','D_fmort','D_rh1','D_fmet','D_rh2','D_FIN','D_FOUT','D_FCIRC','EFIRE_luc','FMORT_luc','RH1_luc','FMET_luc','RH2_luc','EHWP1_luc','EHWP2_luc','EHWP3_luc']
+    if plot is 'all' or plot is 'CH4' or 'CH4' in plot:
+        var_plot += ['D_CH4','D_OHSNK_CH4','D_HVSNK_CH4','D_XSNK_CH4','D_EWET','D_EBB_CH4']
+    if plot is 'all' or plot is 'N2O' or 'N2O' in plot:
+        var_plot += ['D_N2O','D_HVSNK_N2O','D_EBB_N2O']
+    if plot is 'all' or plot is 'O3' or 'O3' in plot:
+        var_plot += ['D_O3t','D_O3s','D_EESC','D_N2O_lag','D_gst']
+    if plot is 'all' or plot is 'AER' or 'AER' in plot:
+        var_plot += ['D_SO4','D_POA','D_BC','D_NO3','D_SOA','D_AERh','RF_SO4','RF_POA','RF_BC','RF_NO3','RF_SOA','RF_cloud']
+    if plot is 'all' or plot is 'clim' or 'clim' in plot:
+        var_plot += ['RF','D_gst','D_gst_CO2','D_gst_CH4','D_gst_SO4','D_gst_BC','D_gyp','RF_CO2','RF_CH4','RF_H2Os','RF_N2O','RF_halo','RF_O3t','RF_O3s','RF_SO4','RF_POA','RF_BC','RF_NO3','RF_SOA','RF_cloud','RF_BCsnow','RF_LCC']
+
+    # save variables
+    var_timeseries = list(set(var_output)|set(var_plot))
+    for var in var_timeseries:
+        # global variables
+        if var in ['D_mld','D_dic','D_pH']\
+        or var in ['OSNK','LSNK','D_OHSNK_CH4','D_HVSNK_CH4','D_XSNK_CH4','D_HVSNK_N2O']\
+        or var in ['D_kOH','D_hv']\
+        or var in ['D_O3t','D_EESC','D_O3s','D_SO4','D_POA','D_BC','D_NO3','D_SOA','D_AERh']\
+        or var in ['D_CO2','D_CH4','D_CH4_lag','D_CH4_fugi_lag','D_N2O','D_N2O_lag']\
+        or var in ['RF','RF_warm','RF_atm','RF_CO2','RF_CO2_from_CH4','RF_CH4','RF_H2Os','RF_N2O','RF_halo','RF_O3t','RF_O3s','RF_SO4','RF_POA','RF_BC','RF_NO3','RF_SOA','RF_cloud','RF_BCsnow','RF_LCC']\
+        or var in ['D_gst','D_gst_CO2','D_gst_CH4','D_gst_SO4','D_gst_BC','D_sst','D_gyp','D_OHC']:
+            exec(var+'_t = np.zeros([ind_final+1],dtype=dty)')
+        # (region) variables
+        if var in ['ELUC','D_AWET','D_EWET','D_ewet','D_EBB_CO2','D_EBB_CH4','D_EBB_N2O','D_EBB_NOX','D_EBB_CO','D_EBB_VOC','D_EBB_SO2','D_EBB_NH3','D_EBB_OC','D_EBB_BC','D_lst','D_lyp']:
+            exec(var+'_t = np.zeros([ind_final+1,nb_regionI],dtype=dty)')        
+        # (region)*(biome) variables
+        if var in ['D_AREA','D_npp','D_efire','D_fmort','D_rh1','D_fmet','D_rh2','D_cveg','D_csoil1','D_csoil2']:
+            exec(var+'_t = np.zeros([ind_final+1,nb_regionI,nb_biome],dtype=dty)')
+        # (region)*(biome)*(biome)*(age) variables
+        if var in ['EFIRE_luc','FMORT_luc','RH1_luc','FMET_luc','RH2_luc','EHWP1_luc','EHWP2_luc','EHWP3_luc']\
+        or var in ['CVEG_luc','CSOIL1_luc','CSOIL2_luc','CHWP1_luc','CHWP2_luc','CHWP3_luc']:
+            exec(var+'_t = np.zeros([ind_final+1,nb_regionI,nb_biome,nb_biome,ind_final+1],dtype=dty)')        
+        # (obox) variables
+        if var in ['D_FIN','D_FOUT','D_FCIRC','D_CSURF']:
+            exec(var+'_t = np.zeros([ind_final+1,nb_obox],dtype=dty)')
+        # (species) variables
+        if var in ['D_HFC','D_HFC_lag','D_OHSNK_HFC','D_HVSNK_HFC','D_XSNK_HFC']:
+            exec(var+'_t = np.zeros([ind_final+1,nb_HFC],dtype=dty)')
+        if var in ['D_PFC','D_PFC_lag','D_OHSNK_PFC','D_HVSNK_PFC','D_XSNK_PFC']:
+            exec(var+'_t = np.zeros([ind_final+1,nb_PFC],dtype=dty)')
+        if var in ['D_ODS','D_ODS_lag','D_OHSNK_ODS','D_HVSNK_ODS','D_XSNK_ODS']:
+            exec(var+'_t = np.zeros([ind_final+1,nb_ODS],dtype=dty)')
+
+    # run variables
+    # ocean
+    D_dic = np.array([0],dtype=dty)
+    D_CSURF = np.zeros([nb_obox],dtype=dty)
+    # land
+    for var in ['D_AREA','D_cveg','D_csoil1','D_csoil2']:
+        exec(var+' = np.zeros([nb_regionI,nb_biome],dtype=dty)')
+    # land-use
+    for var in ['CVEG_luc','CSOIL1_luc','CSOIL2_luc','CHWP1_luc','CHWP2_luc','CHWP3_luc']:
+        exec(var+' = np.zeros([nb_regionI,nb_biome,nb_biome,ind_final+1],dtype=dty)')
+    # atmosphere
+    for var in ['D_CO2','D_CO2_from_CH4','D_CH4','D_CH4_lag','D_CH4_fugi','D_CH4_fugi_lag','D_N2O','D_N2O_lag','D_EESC','D_O3s']:
+        exec(var+' = np.array([0],dtype=dty)')
+    for var in ['D_HFC','D_HFC_lag','D_PFC','D_PFC_lag','D_ODS','D_ODS_lag']:
+        exec(var+' = np.zeros([nb_'+var[2:2+3]+'],dtype=dty)')
+    # climate
+    for var in ['D_gst','D_gst_CO2','D_gst_CH4','D_gst_SO4','D_gst_BC','D_gst0','D_gst0_CO2','D_gst0_CH4','D_gst0_SO4','D_gst0_BC','D_sst','D_gyp','D_OHC']:
+        exec(var+' = np.array([0],dtype=dty)')
+    for var in ['D_lst','D_lyp']:
+        exec(var+' = np.zeros([nb_regionI],dtype=dty)')
+
+    #=======
+    # B. RUN
+    #=======
+    
+    for t in range(1,ind_final+1):
+
+        #=======
+        # B.1. Timestep forcing from Imaclim-R
+        #======= 
+        if t >= 309 and do_dialog_withImaclim:
+            print "Doing year: ", t
+            imaclim_ready = False
+            while not imaclim_ready:
+              if os.path.isfile(path_dialog_withImaclim+"/imaclim"+str(t-300+2001)):
+                imaclim_ready = True
+              else:
+                time.sleep(5) 
+                print "Waiting for Imaclim-R"
+              # safety check
+              if not os.path.exists( path_dialog_withImaclim+"/"):
+                cdxddcvdvdzfopnncpzn=dknqpnejnfhbchbepijpqsj
+            EFF_Im_t = np.genfromtxt( path_dialog_withImaclim+"/imaclim_EFF"+str(t-300+2001), delimiter='|')
+            EFF[t,1:] = EFF_Im_t
+            EN2O_Im_t = np.genfromtxt( path_dialog_withImaclim+"/imaclim_n2o"+str(t-300+2001), delimiter='|')
+            #EN2O[t,1:] = EN2O_Im_t
+            ECH4_Im_t = np.genfromtxt( path_dialog_withImaclim+"/imaclim_ch4"+str(t-300+2001), delimiter='|')
+            #ECH4[t,1:] = ECH4_Im_t
+            # LUC
+            LUC[t,:,:,:] = 0 * LUC[t,:,:,:]
+            for b1 in bio:
+                for b2 in bio:
+                    if os.path.isfile(path_dialog_withImaclim+"/imaclim_"+b1+"2"+b2+str(t-300+2001)):
+                        data_tpt = np.genfromtxt( path_dialog_withImaclim+"/imaclim_"+b1+"2"+b2+str(t-300+2001), delimiter='|')
+                        LUC[t,1:, biome_index[b1], biome_index[b2]] = data_tpt
+                        np.savetxt(path_dialog_withImaclim+"/oscarfound"+b1+"2"+b2+str(t-300+2001), 1, delimiter="|")
+
+        #=======
+        # B.2. Compute 100-yr IRF
+        #======= 
+        listsav=['EFF', 'ECH4', 'ECH4_fugi', 'EN2O', 'LUC', 'HARV', 'SHIFT', 'EHFC', 'EPFC', 'EODS',\
+        'ENOX', 'ECO', 'EVOC', 'ESO2', 'ENH3', 'EOC', 'EBC', \
+        'D_CSURF', 'D_dic', \
+        'CVEG_luc', 'CSOIL2_luc', 'CSOIL1_luc', 'CHWP1_luc', 'CHWP2_luc', 'CHWP3_luc', \
+        'D_EESC', 'D_O3s', \
+        'D_CH4_fugi', 'D_CO2_from_CH4', \
+        'D_CH4_fugi_lag','D_CH4_lag','D_N2O_lag','D_HFC_lag','D_PFC_lag','D_ODS_lag', \
+        'D_CO2', 'D_CH4', 'D_N2O', 'D_HFC', 'D_PFC', 'D_ODS', \
+        'D_gst', 'D_gst_CO2', 'D_gst_CH4', 'D_gst_SO4', 'D_gst_BC', 'D_gst0', 'D_gst0_CO2', 'D_gst0_CH4', 'D_gst0_SO4', 'D_gst0_BC', 'D_sst', 'D_lst', 'D_gyp', 'D_lyp', 'D_OHC']
+        for elt in listsav:
+            str2exec=elt+'sav =copy.deepcopy('+ elt + ')'
+            exec( str2exec)
+        str_oscar_exec = "OSCAR_lite_onestep( p, fT, t, 100, \
+        EFF, ECH4, ECH4_fugi, EN2O, LUC, HARV, SHIFT, EHFC, EPFC, EODS,\
+        ENOX, ECO, EVOC, ESO2, ENH3, EOC, EBC, \
+        RFcon, RFvolc, RFsolar, force_CO2, force_GHG, force_halo, force_RF, force_RFs, force_clim, \
+        D_CSURF, D_dic, \
+        D_AREA, D_cveg, D_csoil1, D_csoil2, \
+        CVEG_luc, CSOIL2_luc, CSOIL1_luc, CHWP1_luc, CHWP2_luc, CHWP3_luc, \
+        D_EESC, D_O3s, \
+        D_CH4_fugi, D_CO2_from_CH4, \
+        D_CH4_fugi_lag,D_CH4_lag,D_N2O_lag,D_HFC_lag,D_PFC_lag,D_ODS_lag, \
+        D_CO2, D_CH4, D_N2O, D_HFC, D_PFC, D_ODS, \
+        D_gst, D_gst_CO2, D_gst_CH4, D_gst_SO4, D_gst_BC, D_gst0, D_gst0_CO2, D_gst0_CH4, D_gst0_SO4, D_gst0_BC, D_sst, D_lst, D_gyp, D_lyp, D_OHC, irf)"
+
+        #if (t >= 299) and (t <= 400) and (t % 10 == 0) and do_dialog_withImaclim and do_IRF:
+        if (t >= 299) and (t <= 304) and (t % 10 == 0) and do_dialog_withImaclim and do_IRF:
+            # Reference
+            irf='_ref'#+str(t)
+            exec(str_oscar_exec)
+            for elt in listsav:
+                exec(elt+' =copy.deepcopy('+ elt + 'sav)')
+            #EFF_sav = copy.deepcopy( EFF)
+            # CO2 - per region
+            for reg in range(nb_regionI-1):
+            #for reg in range(100):
+                EFF[t,reg+1] = EFF[t,reg+1] + 1
+                irf='_CO2_' + str(reg+1)
+                exec(str_oscar_exec)
+                for elt in listsav:
+                    exec(elt+' =copy.deepcopy('+ elt + 'sav)')
+            #    EFF = copy.deepcopy(EFF_sav)
+            # Land-use
+            #LUC_sav = copy.deepcopy( LUC)
+            for b1, b2 in [ ['gra', 'for'], ['cro', 'pas'], ['pas','gra']]:
+                for reg in range(nb_regionI-1):
+                     LUC[t,reg+1, biome_index[b1], biome_index[b2]] = LUC[t,reg, biome_index[b1], biome_index[b2]] + 1
+                     irf='_LUC_'+b1+'_'+b2+'_' + str(reg+1)
+                     exec(str_oscar_exec)
+                     #LUC = copy.deepcopy( LUC_sav)
+                     for elt in listsav:
+                         exec(elt+' =copy.deepcopy('+ elt + 'sav)')
+            np.savetxt(path_dialog_withImaclim+"/oscarfirstIRFdone", RF_warm, delimiter="|")
+        #=======
+        # B.3. Actuel timestep
+        #======= 
+        D_FIN, D_FOUT, D_FCIRC, OSNK, D_CSURF, D_dic, D_AREA, D_AWET, D_k_igni, D_k_rho, D_npp, D_efire, D_fmort, D_rh1, D_fmet, D_rh2, D_ewet, LSNK, D_EWET, D_cveg, D_csoil1, D_csoil2, \
+        CVEG_luc, CSOIL2_luc, CSOIL1_luc, CHWP1_luc, CHWP2_luc, CHWP3_luc, \
+NPP_luc, EFIRE_luc, FMORT_luc, RH1_luc, FMET_luc, RH2_luc, EHWP1_luc, EHWP2_luc, EHWP3_luc, ELUC, \
+        D_EBB_CO2,D_EBB_CH4,D_EBB_N2O,D_EBB_NOX,D_EBB_CO,D_EBB_VOC,D_EBB_SO2,D_EBB_NH3,D_EBB_OC,D_EBB_BC, \
+        D_kOH, D_hv, D_OHSNK_CH4, D_HVSNK_CH4, D_XSNK_CH4, D_OHSNK_CH4_fugi, D_HVSNK_CH4_fugi, D_XSNK_CH4_fugi, D_SNK_CH4_fugi, D_HVSNK_N2O, \
+        D_OHSNK_HFC, D_OHSNK_PFC, D_OHSNK_ODS, \
+        D_HVSNK_HFC, D_HVSNK_PFC, D_HVSNK_ODS, \
+        D_XSNK_HFC, D_XSNK_PFC, D_XSNK_ODS, \
+        D_O3t, D_EESC, D_O3s, D_SO4, D_POA, D_BC, D_NO3, D_SOA, D_DUST, D_SALT, D_AERh, \
+        D_CH4_fugi, D_CO2_from_CH4, \
+        D_CH4_fugi_lag,D_CH4_lag,D_N2O_lag,D_HFC_lag,D_PFC_lag,D_ODS_lag, \
+        D_CO2, D_CH4, D_N2O, D_HFC, D_PFC, D_ODS, \
+        RF_CO2, RF_CO2_from_CH4, RF_CH4, RF_H2Os, RF_N2O, RF_halo, \
+        RF_O3t,RF_O3s,RF_SO4,RF_POA,RF_BC,RF_NO3,RF_SOA,RF_DUST,RF_SALT, \
+        RF_cloud, RF_BCsnow, RF_LCC, \
+        RF, RF_warm, RF_atm, \
+        D_gst, D_gst_CO2, D_gst_CH4, D_gst_SO4, D_gst_BC, D_gst0, D_gst0_CO2, D_gst0_CH4, D_gst0_SO4, D_gst0_BC, D_sst, D_lst, D_gyp, D_lyp, D_OHC, D_pH \
+                = OSCAR_lite_onestep( p, fT, t, 1, \
+        EFF, ECH4, ECH4_fugi, EN2O, LUC, HARV, SHIFT, EHFC, EPFC, EODS,\
+        ENOX, ECO, EVOC, ESO2, ENH3, EOC, EBC, \
+        RFcon, RFvolc, RFsolar, force_CO2, force_GHG, force_halo, force_RF, force_RFs, force_clim, \
+        D_CSURF, D_dic, \
+        D_AREA, D_cveg, D_csoil1, D_csoil2, \
+        CVEG_luc, CSOIL2_luc, CSOIL1_luc, CHWP1_luc, CHWP2_luc, CHWP3_luc, \
+        D_EESC, D_O3s, \
+        D_CH4_fugi, D_CO2_from_CH4, \
+        D_CH4_fugi_lag,D_CH4_lag,D_N2O_lag,D_HFC_lag,D_PFC_lag,D_ODS_lag, \
+        D_CO2, D_CH4, D_N2O, D_HFC, D_PFC, D_ODS, \
+        D_gst, D_gst_CO2, D_gst_CH4, D_gst_SO4, D_gst_BC, D_gst0, D_gst0_CO2, D_gst0_CH4, D_gst0_SO4, D_gst0_BC, D_sst, D_lst, D_gyp, D_lyp, D_OHC)
         if np.isnan(np.sum(D_CO2))|np.isnan(np.sum(D_CH4))|np.isnan(np.sum(D_gst)):
             for var in var_timeseries:
                 if (t < ind_final):
                     exec(var+'_t[t+1:] = np.nan')
             break
+        
+        # If using with Imaclim-R
+        #print "EFF.shape",EFF.shape
+        #print "RF_warm", RF_warm.shape
+        #print "D_gst", D_gst.shape
+        #print "D_AREA", D_AREA.shape
+        #print "LUC", LUC.shape
+        if t >= 309 and do_dialog_withImaclim:
+             #f=open("../../outputs/ImaclimROscar_dialog/oscar"+str(t-300+2001),"w+")
+             #f.write(D_gst)
+             #f.close()
+             np.savetxt(path_dialog_withImaclim+"/oscar"+str(t-300+2001), RF_warm, delimiter="|")
 
     #===========
     # C. FIGURES
diff -u OSCAR-2.2.2/OSCAR.py OSCAR/OSCAR.py
--- OSCAR-2.2.2/OSCAR.py	2018-07-11 10:13:19.000000000 +0200
+++ OSCAR/OSCAR.py	2020-05-19 17:41:02.892087152 +0200
@@ -29,6 +29,17 @@
 from scipy.special import gammainc
 from matplotlib.font_manager import FontProperties
 
+##################################################
+#   0. Default from Imaclim
+##################################################
+if not "RCPXX" in globals():
+    RCPXX="RCP2.6"
+
+if not "do_IRF" in globals():
+    do_IRF=False
+
+if not "do_dialog_withImaclim" in globals():
+    do_dialog_withImaclim=False 
 
 ##################################################
 #   1. OPTIONS
@@ -40,7 +51,7 @@
 dty = np.float32                        # precision (float32 or float64)
 
 PI_1750 = True                         # if False simulates the 1700-1750 period
-ind_final = 310                         # ending year of run (+1700)
+ind_final = 500                         # ending year of run (+1700)
 ind_attrib = 0                          # starting year of attribution
 
 attrib_DRIVERS = 'producers'            # [deprecated]
@@ -49,7 +60,9 @@
 attrib_ELUCampli = 'causal'             # [deprecated]
 
 mod_regionI = 'Houghton'                # SRES4 | SRES11 | RECCAP* | Raupach* | Houghton | IMACLIM | Kyoto | RCP5 | RCP10*
+mod_regionI = 'IMACLIM'                # SRES4 | SRES11 | RECCAP* | Raupach* | Houghton | IMACLIM | Kyoto | RCP5 | RCP10*
 mod_regionJ = 'RCP5'                    # SRES4 | SRES11 | RECCAP* | Raupach* | Houghton | IMACLIM | Kyoto | RCP5 | RCP10*
+mod_regionJ = 'IMACLIM'                    # SRES4 | SRES11 | RECCAP* | Raupach* | Houghton | IMACLIM | Kyoto | RCP5 | RCP10*
 mod_sector = ''                         # '' | Time | TimeRCP
 mod_kindFF = 'one'                      # one | CDIAC
 mod_kindLUC = 'one'                     # one | all
@@ -78,7 +91,7 @@
 
 mod_DATAscen = 'trends'                 # raw | offset | smoothX (X in yr) | trends
 
-scen_ALL = ''                           # '' | stop | cst | RCP8.5 | RCP6.0 | RCP4.5 | RCP2.6
+scen_ALL = RCPXX                           # '' | stop | cst | RCP8.5 | RCP6.0 | RCP4.5 | RCP2.6
 
 scen_EFF = 'stop'                       # stop | cst | SRES-A1B | SRES-A1FI | SRES-A1T | SRES-A2 | SRES-B1 | SRES-B2 | RCP8.5 | RCP6.0 | RCP4.5 | RCP2.6
 scen_LULCC = 'stop'                     # stop | cst | RCP8.5 | RCP6.0 | RCP4.5 | RCP2.6
